scripts/translate_pptx_inplace.py

```
@@ -258,75 +271,479 @@ def _chat_create(client, model: str, sys_prompt: str, user_payload: dict, temper
-                depth -= 1
-                if depth == 0:
-                    frag = s[start:i+1]
-                    try:
-                        arr = json.loads(frag)
-                        if isinstance(arr, list) and (expected_len == 0 or len(arr) == expected_len):
-                            return arr
-                    except Exception:
-                        pass
-                    break
+    dec = json.JSONDecoder()
+    in_str = esc = False; i = 0; n = len(s)
```

Ruff (0.12.2)

Multiple statements on one line (semicolon)

(E702)